packages/fsapp/src/beta-app/modal/modal.context.tsx

```diff
@@ -87,8 +87,15 @@ export const ModalProvider: FC<ModalProviderProps> = ({ children, screenWrap })
                   id,
                   name: modal.definitionId,
                   options: {
-                    modal: modal.options,
-                    topBar: modal.topBarOptions
+                    ...modal.options?.navigationOptions,
+                    modal: {
+                      ...modal.options?.navigationOptions?.modal,
+                      ...modal.options
+                    },
+                    topBar: {
+                      ...modal.options?.navigationOptions?.topBar,
+                      ...modal.topBarOptions
+                    }
```

Comment:
It feels a bit weird to include so many extra values in `options.modal`, ie `navigationOptions`.
This will behave as you expect it to, there will just be extra values in there that are unused.
That said we were already doing that with the web props so probably not a big deal.